pkg/list/lister.go

```diff
@@ -206,9 +208,9 @@ func (l *lister) fetchRepoAndInstalled(ctx context.Context, options ListOptions,
 	// TODO what if a package is namespaced in one repository, and with the same name cluster scoped in another??
 
 	resultLs := make([]result, 0)
-	for i, indexPackage := range index.Packages {
+	for _, indexPackage := range index.Packages {
 		res := result{
-			IndexItem: &index.Packages[i],
+			IndexItem: &indexPackage,
```

Comment:
I think this change should be reverted. 

Even though pointing to the loop variable is not an error anymore in Go 1.22, I think it's still good practice and easier to understand to point to the array element.